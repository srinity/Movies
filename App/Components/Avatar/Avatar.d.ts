import { Component } from 'react'

export interface AvatarData {
  /**
   * The id of the movie.
   *
   */
  id: number

  /**
   * The image of the avatar/actor/actress.
   *
   */
  image: string

  /**
   * The name of the avatar/actor/actress.
   *
   */
  name: string
}

export interface AvatarProps extends AvatarData {}

export default class Avatar extends Component<AvatarProps> {}
