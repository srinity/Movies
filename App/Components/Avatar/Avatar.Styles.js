import { StyleSheet } from 'react-native'

import { Colors, Fonts } from '../../Theme'

export default StyleSheet.create({
  containerStyle: {
    alignItems: 'center',
    marginRight: 15,
    width: 85
  },
  imageStyle: {
    height: 75,
    width: 75,
    borderRadius: 37.5,
    overflow: 'hidden'
  },
  titleStyle: {
    color: Colors.blackColor,
    fontSize: Fonts.h5,
    marginTop: 7,
    textAlign: 'center'
  }
})
