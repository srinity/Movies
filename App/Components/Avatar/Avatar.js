import React from 'react'
import { View, Image, Text } from 'react-native'
import PropTypes from 'prop-types'
import { isNil as _isNil } from 'lodash'

import { Images } from '../../Theme'

import styles from './Avatar.Styles'

function Avatar({ image, name, ...props }) {
  return (
    <View style={styles.containerStyle}>
      <Image
        source={_isNil(image) ? Images.avatar : { uri: image }}
        style={styles.imageStyle}
        resizeMode="cover"
      />

      <Text style={styles.titleStyle}>{name}</Text>
    </View>
  )
}

Avatar.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string.isRequired
}

export default Avatar
