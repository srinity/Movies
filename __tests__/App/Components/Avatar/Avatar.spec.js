import React from 'react'
import renderer from 'react-test-renderer'

import Avatar from '../../../../App/Components/Avatar/Avatar'

describe('Avatar Component', () => {
  it('should render correctly', () => {
    const avatar = renderer.create(
      <Avatar
        image="https://www.themoviedb.org/t/p/w1280/rmIYKVdYT60zfsqfjNuB71f1y82.jpg"
        name="LeBron James"
      />
    )
    expect(avatar).toMatchSnapshot()
  })
})
