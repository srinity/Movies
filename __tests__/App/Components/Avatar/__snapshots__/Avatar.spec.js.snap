// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginRight": 15,
      "width": 85,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://www.themoviedb.org/t/p/w1280/rmIYKVdYT60zfsqfjNuB71f1y82.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 37.5,
        "height": 75,
        "overflow": "hidden",
        "width": 75,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 14,
        "marginTop": 7,
        "textAlign": "center",
      }
    }
  >
    LeBron James
  </Text>
</View>
`;
